module/VuFind/src/VuFind/I18n/Initializer.php

```diff
@@ -0,0 +1,289 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Translator
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\I18n;
+
+use VuFind\Cookie\CookieManager;
+use VuFind\I18n\Translator\Loader\ExtendedIni;
+use Zend\Cache\Storage\StorageInterface;
+use Zend\Config\Config;
+use Zend\Http\PhpEnvironment\Request;
+use Zend\I18n\Translator\Translator as I18nTranslator;
+use Zend\Mvc\I18n\Translator as MvcTranslator;
+use Zend\View\Model\ViewModel;
+
+/**
+ * Handles I18n initialization.
+ *
+ * @category VuFind
+ * @package  Translator
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class Initializer
+{
+    /**
+     * Translator.
+     *
+     * @var I18nTranslator
+     */
+    protected $translator;
+
+    /**
+     * Language base directories.
+     *
+     * @var string[]
+     */
+    protected $baseDirs = [];
+
+    /**
+     * File loader.
+     *
+     * @var ExtendedIni
+     */
+    protected $loader;
+
+    /**
+     * Initializer constructor.
+     *
+     * @param Request          $request       Request
+     * @param Config           $config        Configuration
+     * @param CookieManager    $cookies       Cookie manager
+     * @param StorageInterface $cache         Language cache
+     * @param MvcTranslator    $mvcTranslator Translator
+     * @param ViewModel        $viewModel     ViewModel
+     */
+    public function __construct(
+        Request $request,
+        Config $config,
+        CookieManager $cookies,
+        StorageInterface $cache,
+        MvcTranslator $mvcTranslator,
+        ViewModel $viewModel
+    ) {
+        $language = $this->detectLanguage($config, $request, $cookies);
+
+        $this->translator = $mvcTranslator->getTranslator();
+        $this->loader = $this->translator->getPluginManager()
+            ->get(ExtendedIni::class);
+
+        $this->translator->setLocale($language);
+        $this->translator->setCache($cache);
+
+        $this->loader->setFallbacks($this->parseFallbacks($config));
```

Comment:
Parse language fallbacks as provided by configuration and pass them to the loader.